dm/ctl/master/operate_task.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package master
+
+import (
+	"context"
+	"errors"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pingcap/dm/dm/ctl/common"
+	"github.com/pingcap/dm/dm/pb"
+)
+
+const (
+	batchSizeFlag    = "batch-size"
+	defaultBatchSize = 5
+)
+
+type batchTaskResult struct {
+	Result       bool                 `json:"result"`
+	Msg          string               `json:"msg"`
+	BatchResults []*operateTaskResult `json:"batchResults"`
+}
+
+type operateTaskResult struct {
+	TaskName string                     `json:"taskName"`
+	Op       string                     `json:"op"`
+	Result   bool                       `json:"result"`
+	Msg      string                     `json:"msg"`
+	Sources  []*pb.CommonWorkerResponse `json:"sources"`
+}
+
+func operateTaskFunc(taskOp pb.TaskOp, cmd *cobra.Command) error {
+	argLen := len(cmd.Flags().Args())
+	if argLen == 0 {
+		// may want to operate tasks bound to a source
+		return operateSourceTaskFunc(taskOp, cmd)
+	} else if argLen > 1 {
+		// can pass at most one task-name/task-conf
+		cmd.SetOut(os.Stdout)
+		common.PrintCmdUsage(cmd)
+		return errors.New("please check output to see error")
+	}
+
+	name := common.GetTaskNameFromArgOrFile(cmd.Flags().Arg(0))
+	sources, err := common.GetSourceArgs(cmd)
+	if err != nil {
+		return err
+	}
+
+	resp, err := common.OperateTask(taskOp, name, sources)
+	if err != nil {
+		common.PrintLinesf("can not %s task %s", strings.ToLower(taskOp.String()), name)
+		return err
+	}
+
+	common.PrettyPrintResponse(resp)
+	return nil
+}
+
+func addOperateSourceTaskFlags(cmd *cobra.Command) {
+	// control workload to dm-cluster for sources with large number of tasks.
+	cmd.Flags().Int(batchSizeFlag, defaultBatchSize, "batch size")
```

Comment:
Shall we remove this parameter by automatically providing a proper value?🤔

Reply:
> Shall we remove this parameter by automatically providing a proper value?🤔
now it's a simple rule, default to 5, but can be adjusted manually
the max number of tasks related to a single source ever met on production env is 10+, so it seems enough for most cases.
maybe we can enhance it later?